bftengine/src/preprocessor/RequestProcessingInfo.cpp

```diff
@@ -65,20 +65,26 @@ PreProcessingResult RequestProcessingInfo::getPreProcessingConsensusResult() con
   if (maxNumOfEqualHashes >= numOfRequiredEqualReplies_) {
     if (itOfChosenHash->first == myPreProcessResultHash_)
       return COMPLETE;  // Pre-execution consensus reached
-    else {
-      // Primary replica calculated hash is different from hash that passed consensus => we don't have correct
-      // pre-processed results. The request could be cancelled, retried or executed without pre-processing.
+    else if (myPreProcessResultLen_ != 0) {
+      // Primary replica calculated hash is different from a hash that passed pre-execution consensus => we don't have
+      // correct pre-processed results. Let's launch a pre-processing retry.
```

Comment:
I cannot understand the logical mapping between the `if (myPreProcessResultLen_ != 0)` and the comment below...

Reply:
Once the primary pre-executed the request, it calculates its hash and saves it in myPreProcessResultHash_. 
myPreProcessResultLen_ stores myPreProcessResultHash_ length.

Reply:
So, it's either 0 or 1 ?

Reply:
In fact, it contains a size of myPreProcessResultHash_.

Reply:
and what and how many elements can be contained in `myPreProcessResultHash_`?

Reply:
What elements? RequestProcessingInfo object is created per request.

Reply:
Ahhh, never mind, myPreProcessResultHash_ is a single hash :) Got it now. We only retry if the myPreProcessResultHash_ is non-empty.

Reply:
Exactly!